language-design.md

```diff
@@ -74,7 +122,11 @@ the commenters?
+
+This is what I think should be the case for programming languages. Most PLs should not cater to the lowest denominator (and read like a self-contained book). It is already hard enough to navigate code and making it more verbost will not help that. However, creators of PLs should strive to encourage readability and clarity so that it is easy for programmers who are familiar with the language to understand each other's code. If code in a language easily becomes too dense or "WORN" (write once, read never), then its maintainability (and therefore efficiency) will be severely impacted.
 
+I think [Strengthiness' comment](https://www.fastcodesign.com/1665735/why-arent-computer-programming-languages-designed-better#comment-93e75ca0-9b5b-11e4-a3e4-4930d33952b2) does a good job of summarizing my opinion on this issue: "...there's a reason why programming languages professional programmers use are precise and terse (-ish…). A language like Quorum may be good for learning programming concepts, but as an everyday tool for a professional programmer it would be so overly verbose as to just slow one down. Our languages may be esoteric and the tools hard to master, but show me any other technical profession where tools are straightforward to learn and mastery comes easily." I also agree with [Mjaksen](https://www.fastcodesign.com/1665735/why-arent-computer-programming-languages-designed-better#comment-f3521300-7b3d-11e2-9387-9794fc737ae8) that even if we tried to make programming languages more approachable, it would cause more complications, leading them to conclude "there \*really\* is no pleasing everyone" (which Joshua Bloch agrees with when it comes to APIs). [Simon](https://www.fastcodesign.com/1665735/why-arent-computer-programming-languages-designed-better#comment-95ed5400-5bc1-11e1-921b-9794fc737ae8) also makes an excellent point that reafirms what Strengthiness wrote above: "The issue with writing software is not being able to program; it's being able to program \*well\*. There are a number of languages out there which have been specifically designed to be easier to write for novice programmers. What generally ends up happening is that novices then end up writing software, and the quality of the end product suffers as a result. I'm not knocking novices -- we were all novices at one point -- but I guess most companies have got somewhere an Access database system or simple VB6 program that they rely on for some task but which is woefully inadequate, or worse, a security minefield, because it was written by someone who didn't really understand the technical issues." Perhaps the problem is not that programming languages are too hard, but that there currently isn't enough technical training on how to use them (and especially on how to use them well).
 
```

Comment:
There's also a distinction between a professional programmer, who are expected to write software for companies and for whom understanding issues such as security are important, and other people who program. The comments you quoted provide the perspective of the professional programmer, but is that the only perspective? (Probably not.) Is it the most important? (Depends?)